### PR TITLE
deps: update jsonrpsee to v0.22.5

### PR DESCRIPTION
**Pull request by koushiro**
use `server-core` feature instead of `server` feature when defining the rpc api

### CODE REVIEW

File: substrate/client/consensus/babe/rpc/Cargo.toml [hunk 1]
```diff
@@ -16,7 +16,7 @@ workspace = true
 targets = ["x86_64-unknown-linux-gnu"]
 
 [dependencies]
-jsonrpsee = { version = "0.22", features = ["client-core", "macros", "server"] }
+jsonrpsee = { version = "0.22", features = ["client-core", "server-core", "macros"] }
```
Comment (jasl) [inline substrate/client/consensus/babe/rpc/Cargo.toml:19]: If the `server-core` feature fails to compile before 0.22.5, maybe you need to point the version to 0.22.5.

Comment (koushiro) [reply substrate/client/consensus/babe/rpc/Cargo.toml]: Yeah, for downstream users, it's neceessary

**Comment by jasl** [comment]
Cool, it should be transparency for downstream but reduce several dependencies

**Comment by skunert** [comment]
In general LGTM, just curious why this needs 22.5? Could not find anything in the jsonrpsee changelogs that seems relevant.

**Review by bkchr — APPROVED** [review]

**Review by skunert — APPROVED** [review]

**Pull request merged**